### ISSUE TITLE
Typescript error when composing a React Component with `forwardRef`

### ISSUE THREAD

**Issue by peduarte** [modulz/stitches#71]
Repro: https://codesandbox.io/s/stitches-ts-error-with-reactforwardref-s37te

<img width="770" alt="CleanShot 2020-08-05 at 16 55 58@2x" src="https://user-images.githubusercontent.com/372831/89428321-8eae0400-d73c-11ea-822c-01cf02c49771.png">

@christianalfoni wonder if you have come across this one before?
From Christian:
> Ah, yeah! I believe we are using the wrong component typing for that :slight_smile: but want to take another round on the typing. Now that styled components is faster I am thinking of just copying that, cause they have been iterating on this for a while and there seems to be a lot of small edge cases :slight_smile: and then rather continue optimizing from there, so it can be shared with styled components

**Comment by peduarte** [comment]
@hadihallak maybe you can take a look too?

**Comment by hadihallak** [comment]
**Added a couple more related usecases that produce similar typing errors even when not using forwardRef:**
Codesandbox: https://codesandbox.io/s/stitches-ts-error-ulycv
```tsx
import React from 'react'
import { styled } from '../utils/stitches'

const variants = { variant: { red: {} } } as const
// Button primitive with forwardref
const ButtonPrimitiveOne = React.forwardRef<HTMLButtonElement, React.ComponentPropsWithRef<'button'>>(
  (props, ref) => <button {...props} ref={ref} />,
)

// having a children prop on a normal component will produce the same error as with forwardRef
const ButtonPrimitiveTwo = (props: { children: any }) => <button {...props} />

// styled button primitive with forward ref:
// will give the same error as the one Pedro posted
const StyledButtonWithForwardRef = styled(ButtonPrimitiveOne, {}, variants)

// styled button primitive without forward ref:
// will give the same error as the one Pedro posted
const StyledButtonWithoutForwardRef = styled(ButtonPrimitiveTwo, {}, variants)

// variants will work here but we're casting everything
const StyledButtonPrimitiveWithWorkingVariants = styled((ButtonPrimitiveOne as any) as 'button', {}, variants)

export default function App() {
  return (
    <div className="App">
      {/** Variants not typed */}
      <StyledButtonWithForwardRef variant="red">Hello</StyledButtonWithForwardRef>
      {/** Variants not typed */}
      <StyledButtonWithoutForwardRef variant="red">Hello</StyledButtonWithoutForwardRef>
      {/** Variants typed but we're casting everything */}
      <StyledButtonPrimitiveWithWorkingVariants variant="red">
        Button
      </StyledButtonPrimitiveWithWorkingVariants>
    </div>
  )
}

```

**Comment by peduarte** [comment]
@hadihallak  @christianalfoni Just a reminder this is a high priority one. Without this we cant correctly style our Primitives 👌

**Comment by christianalfoni** [comment]
Okay, I believe this is fixed in `master` now 😄 

https://github.com/modulz/stitches/commit/842a55bb0447cd01c08dd6f267477924fd74f66d

**Comment by hadihallak** [comment]
Fixed in #134
